modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/internal/jaxrs/application/FrontendClayApplication.java

```diff
@@ -148,6 +153,41 @@ public Response getClayDataSetData(
 		return singletons;
 	}
 
+	@Path("/data-set/{id}/set-active-view/{activeViewName}")
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response setActiveClayDataSetView(
+		@PathParam("id") String id,
+		@PathParam("activeViewName") String activeViewName,
+		@Context HttpServletRequest httpServletRequest,
+		@Context HttpServletResponse httpServletResponse,
+		@Context UriInfo uriInfo) {
+
+		try {
+			PortalPreferences portalPreferences =
+				PortletPreferencesFactoryUtil.getPortalPreferences(
+					httpServletRequest);
+
+			portalPreferences.setValue(
```

Comment:
Based on where we want to take this... I was wondering if we want to make something more generic already, or it's okay since preferences will be left orphan if we ever change them.

That is should we do something like `setValue(id, "datasetconfig", {activeView: "table"}`. Or even more specific passing in `schema` and the other models we have for this?

So, I'd expect to have a config `json` stored per `dataset` with its configuration value. Seeing as we will be adding schema, `activeViewName` will probably be overturned by something else more generic. We could of course spread the first-level entries in the configuration, but I guess we'd rather retrieve one preference than several ones.

Reply:
Yeah, I think for now it still makes sense if we store a json with just name in it. Then all we have to do is add schema or anything else in it on the future. I’ll add that as soon as I get my Mac fixed.